src/TLuaInterpreter.cpp

```diff
@@ -8295,6 +8345,116 @@ int TLuaInterpreter::permAlias(lua_State* L)
     return 1;
 }
 
+int TLuaInterpreter::getScript(lua_State* L)
+{
+    int n = lua_gettop(L);
+    int pos = 1;
+
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "getScriptScript: bad argument #1 type (script name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+    QString name = QString::fromUtf8(lua_tostring(L, 1));
+
+    if (n > 1) {
+        if (!lua_isnumber(L, 2)) {
+            lua_pushfstring(L, "getScriptScript: bad argument #2 type (script position as number expected, got %s!)", luaL_typename(L, 2));
+            return lua_error(L);
+        }
+        pos = lua_tonumber(L, 2);
+    }
+    Host& host = getHostFromLua(L);
+    pos--;
+    auto ids = host.getScriptUnit()->findScriptId(name);
+    auto pS = host.getScriptUnit()->getScript(ids[pos]);
+
+    if (!pS) {
+        lua_pushnil(L);
+        lua_pushstring(L, QStringLiteral("script \"%1\" not found").arg(name).toUtf8().constData());
+        return 2;
+    }
+
+    int id = pS->getID();
+    lua_pushstring(L, pS->getScript().toUtf8().constData());
+    lua_pushnumber(L, id);
+    return 2;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setScript
+int TLuaInterpreter::setScript(lua_State* L)
```

Comment:
This doesn't seem to work for new or existing scripts - just returns `-1`.

Reply:
Same here. 
If the script is not compiling successfully it will not change the existing script.
It should return also a string with the error message?

Reply:
ohh for sure